### PR TITLE
[4.1] RavenDB-11771 AlertRaised/DatabaseTopologyWarning Write Transaction e…

### PR DESCRIPTION
**Pull request by karmeli87**
…very 0.5 second

Open write transaction only if needed.

### CODE REVIEW

File: src/Raven.Server/NotificationCenter/NotificationsStorage.cs [hunk 1]
```diff
@@ -80,30 +80,32 @@ public void Initialize(StorageEnvironment environment, TransactionContextPool co
...
+                    }
                 }
 
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Saving notification '{notification.Id}'.");
+
                 using (var json = context.ReadObject(notification.ToJson(), "notification", BlittableJsonDocumentBuilder.UsageMode.ToDisk))
+                using (var tx = context.OpenWriteTransaction())
                 {
                     Store(context.GetLazyString(notification.Id), notification.CreatedAt, postponeUntil, json, tx);
```
Comment (ayende) [inline src/Raven.Server/NotificationCenter/NotificationsStorage.cs:106]: Need to do the check here again, you might have had another tx come in the meantime.

Comment (karmeli87) [reply src/Raven.Server/NotificationCenter/NotificationsStorage.cs:106]: In any case someone will be overwritten here.. We identify the notification by the Id, which should be unique to the case.
But when 2 notifications with the same Id are concurrently stored, I'm not sure if we should care which one will be written.

Comment (ayende) [reply src/Raven.Server/NotificationCenter/NotificationsStorage.cs:106]: OK

File: src/Raven.Server/NotificationCenter/NotificationsStorage.cs [hunk 2]
```diff
@@ -80,30 +80,32 @@ public void Initialize(StorageEnvironment environment, TransactionContextPool co
...
+                            return false;
 
-                    if (existing?.PostponedUntil != null && existing.PostponedUntil.Value > SystemTime.UtcNow)
-                        postponeUntil = existing.PostponedUntil;
+                        if (existing?.PostponedUntil != null && existing.PostponedUntil.Value > SystemTime.UtcNow)
+                            postponeUntil = existing.PostponedUntil;
+                    }
                 }
 
+                if (Logger.IsInfoEnabled)
```
Comment (aviviadi) [inline src/Raven.Server/NotificationCenter/NotificationsStorage.cs:100]: what happens if something fails above? (should we at least try to log it)

Comment (karmeli87) [reply src/Raven.Server/NotificationCenter/NotificationsStorage.cs:100]: See here: https://github.com/ravendb/ravendb/blob/cbcffe3c89b8b37b57f34ec6510854079f895dee/src/Raven.Server/NotificationCenter/NotificationCenter.cs#L68

**Pull request merged**